src/utils/context_utils.py

```diff
@@ -96,6 +97,24 @@ def get_story_text_generator(
             },
         )
 
+        if server_settings.allow_backup_story_models:
+            providers = [
+                lambda: generator
+            ]
+            for backup_model_name in open_ai_models:
```

Comment:
There's an implicit understanding that `open_ai_models` is roughly prioritized here.